packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -72,6 +67,16 @@ export function createRootHooks<
   const TRPCProvider: TRPCProvider<TRouter, TSSRContext> = (props) => {
     const { abortOnUnmount = false, client, queryClient, ssrContext } = props;
     const [ssrState, setSSRState] = useState<SSRState>(props.ssrState ?? false);
+
+    const queryFns = useMemo(
+      () =>
+        createUtilityFunctions({
+          client,
+          queryClient,
+        }),
+      [client, queryClient],
+    );
```

Comment:
Could you memo the whole `value`? then we get #4773 for free

```ts
  const fns = useMemo(
    () => createTRPCQueryUtils({ queryClient, client }),
    [queryClient, client],
  );

  const value: ProviderContext = useMemo(
    () => ({
      ...fns,
      abortOnUnmount,
      queryClient,
      client,
      ssrContext: ssrContext ?? null,
      ssrState,
    }),
    [abortOnUnmount, client, queryClient, ssrContext, ssrState, fns],
  );
```

Reply:
Cool, done